### PR TITLE
fix: post options menu state

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes
- When viewing a post page, and you try to follow/unfollow certain tags/sources/settings, the values do not update when you do any action.
- In this PR, we now consider the current state of the user's preference to reflect better the actual status.
- Introduced a new settings function to update the value based on the desired state. This was due to a bug encountered were the value is not getting updated after clicking undo (because of the memoized function).

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-2057 #done

### CODE REVIEW

File: packages/shared/src/hooks/useFeedSettings.ts [hunk 1]
```diff
@@ -53,14 +54,29 @@ export default function useFeedSettings({
   );
 
   const { tagsCategories, feedSettings, advancedSettings } = feedQuery;
+  const checkEnabledSettings = useCallback(
+    (value: number) => {
```
Comment (rebelchris) [inline packages/shared/src/hooks/useFeedSettings.ts:58]: Value is the advancedSettingsId?
I would maybe call it that or `id` just so it's clear.

File: packages/shared/src/hooks/useFeedSettings.ts [hunk 2]
```diff
@@ -53,14 +54,29 @@ export default function useFeedSettings({
   );
 
   const { tagsCategories, feedSettings, advancedSettings } = feedQuery;
+  const checkEnabledSettings = useCallback(
```
Comment (rebelchris) [inline packages/shared/src/hooks/useFeedSettings.ts:57]: Also the naming contradict it searches one value.
checkSettingsEnabledState or something might be better? (WDYT)

File: packages/shared/src/hooks/feed/useAdvancedSettings.ts [hunk 3]
```diff
@@ -29,34 +30,36 @@ export const useAdvancedSettings = (
...
-      selectedSettings,
-      trackEvent,
-      updateAdvancedSettings,
-      updateAlerts,
-      user,
-    ],
+    [trackEvent, updateAdvancedSettings],
+  );
+
+  // We still need to support this function as this is used for experiments
```
Comment (rebelchris) [inline packages/shared/src/hooks/feed/useAdvancedSettings.ts:49]: We can't migrate both into one?

**Review by rebelchris — COMMENTED** [review]
Nice work, nothing blocking really, but let's wait for 2 reviews :)

File: packages/shared/src/hooks/useFeedSettings.ts (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/hooks/useFeedSettings.ts:58]: Right, I was reserving the `id` on the destructure part, but we can do that.

File: packages/shared/src/hooks/feed/useAdvancedSettings.ts (hunk 3 above)
Comment (sshanzel) [reply packages/shared/src/hooks/feed/useAdvancedSettings.ts:49]: We can, probably. Just not sure if we should do it as we've had two experiments on separate occasions on this part. Now that we have new advanced settings, we might again introduce the same experiment. Wdyt?

File: packages/shared/src/hooks/useFeedSettings.ts (hunk 2 above)
Comment (sshanzel) [reply packages/shared/src/hooks/useFeedSettings.ts:57]: Makes sense. Let's update it.

File: packages/shared/src/hooks/feed/useAdvancedSettings.ts (hunk 3 above)
Comment (rebelchris) [reply packages/shared/src/hooks/feed/useAdvancedSettings.ts:49]: Hmm yeah let's not complicate it for now :) 
We can adjust later as well.

File: packages/shared/src/hooks/useFeedSettings.ts [hunk 4]
```diff
@@ -53,14 +54,31 @@ export default function useFeedSettings({
   );
 
   const { tagsCategories, feedSettings, advancedSettings } = feedQuery;
+  const checkSettingsEnabledState = useCallback(
+    (idParam: number) => {
+      const advancedSetting = advancedSettings?.find(
+        ({ id }) => id === idParam,
+      );
+
+      if (!advancedSettings) {
```
Comment (kopancek) [inline packages/shared/src/hooks/useFeedSettings.ts:63]: I think this should be 
```suggestion
      if (!advancedSetting) {
```

**Review by kopancek — CHANGES_REQUESTED** [review]

File: packages/shared/src/hooks/useFeedSettings.ts (hunk 4 above)
Comment (sshanzel) [reply packages/shared/src/hooks/useFeedSettings.ts:63]: Oh, good catch! 🚀

**Review by kopancek — APPROVED** [review]
Nothing more to add, looks good to me, great job! 👍 :shipit:

File: packages/shared/src/hooks/useFeedSettings.ts (hunk 4 above)
Comment (sshanzel) [reply packages/shared/src/hooks/useFeedSettings.ts]: Maybe a good sign that I should rename the variable name entirely to void confusion.

**Pull request merged**